### PR TITLE
Depreciation cleanups; require c++11

### PR DESCRIPTION
**Pull request by anadon**
@jzmaddock I have a bigger one for you.

Cleaned up a lot of old depreciation warnings, and starting to add logic for coverage CI.

Also, there were a few instances where C++11 was just so easier, and so now BGL requires C++11 support.  As I go forward, this will become more important.

### CODE REVIEW

**Comment by anadon** [comment]
@Glenfe in Slack mentioned that some of the C++11 requirements w/r std::bind1st can be replaced with boost::bind which has C++03 support.  It looks like he's going to review a few of these changes.

File: include/boost/graph/labeled_graph.hpp [hunk 1]
```diff
@@ -386,7 +386,10 @@ class labeled_graph
 
     /** Remove the vertex with the given label. */
     void remove_vertex(Label const& l)
-    { return boost::remove_vertex(vertex(l), _graph); }
+    {
+        // _map.erase(l);
```
Comment (anadon) [inline include/boost/graph/labeled_graph.hpp]: Related to #167 .

**Comment by jeremy-murphy** [comment]
As I have said previously, everyone appreciates effort, but I don't think this is the right way to progress the BGL. The best pull requests are small, not big; they focus on one thing, or one idea across several areas. This is a mixture of what looks like mostly white-space changes and actual code changes, which makes the actual code changes easy to miss. I recommend to just not worry about white-space at all. It's something we can fix automatically with clang-format one day -- no-one wants to review white-space changes, and it's just risky to mix them up with actual material changes.
Also, I don't think changing the base language standard of a library should happen on the whim of a pull request. I'm all for progress, but we also have a community that uses this library and they should probably be consulted with in case we happen to have a lot or a few significant users that rely on '03 compatibility.
Let's hope not, but maybe let's ask first?

**Comment by anadon** [comment]
@jeremy-murphy Glen Fernandes caught me on a number of issues with this commit.  Check back in about a week or so.  I should have it in a shape you find considerably less objectionable.

File: .travis.yml [hunk 2]
```diff
@@ -27,44 +27,6 @@ matrix:
...
-
-    - os: linux
-      env: TOOLSET=gcc COMPILER=g++-4.8 CXXSTD=03,11
-      addons:
-        apt:
-          packages:
-            - g++-4.8
-          sources:
-            - ubuntu-toolchain-r-test
-
```
Comment (jzmaddock) [inline .travis.yml]: Boost in general is still testing with these older compiler, I'm not sure we want to remove all these?

File: example/fr_layout.cpp [hunk 3]
```diff
@@ -16,7 +16,6 @@
 #include <map>
 #include <vector>
 #include <boost/random/linear_congruential.hpp>
-#include <boost/progress.hpp>
```
Comment (jzmaddock) [inline example/fr_layout.cpp]: I know boost/progress.hpp is old stuff, but I assume program output is a lot less pretty without this?  What's the rationale for removing it?

File: example/graph-assoc-types.cpp [hunk 4]
```diff
@@ -10,23 +10,27 @@
...
 using namespace boost;
 
-template < typename Graph > void
-generic_foo(Graph & g)
-{
+template < typename Graph > 
+void
+generic_foo(
+  Graph & g
+){
```
Comment (jzmaddock) [inline example/graph-assoc-types.cpp]: I know I said I'd accept any sensible reformating, but please not mixed up with substantive changes, and IMO this is a horrible format :(

File: example/graphviz.cpp [hunk 5]
```diff
@@ -43,37 +42,41 @@ void test_graph_read_write(const std::string& filename)
...
 }
 
-int test_main(int argc, char* argv[])
-{
+
+int
+main(
+  int argc, 
+  char* argv[]
+){
```
Comment (jzmaddock) [inline example/graphviz.cpp]: Sorry, but that formatting really does grate with me.

File: include/boost/graph/adjacency_list_io.hpp [hunk 6]
```diff
@@ -76,7 +76,7 @@ void get
 }
 
 // get a subset of properties tagged Stag
-template<class Tag, class Value, class Next, 
+template<class Tag, class Value, class Next,
```
Comment (jzmaddock) [inline include/boost/graph/adjacency_list_io.hpp]: Does this file contain anything other than whitespace changes?   It's *very* hard to review stuff like this.... sorry!

File: include/boost/graph/bc_clustering.hpp [hunk 7]
```diff
@@ -147,14 +147,14 @@ betweenness_centrality_clustering(MutableGraph& g, Done done,
...
- */ 
+ */
 template<typename MutableGraph, typename Done>
 void
 betweenness_centrality_clustering(MutableGraph& g, Done done)
 {
   typedef typename Done::centrality_type centrality_type;
   std::vector<centrality_type> edge_centrality(num_edges(g));
-  betweenness_centrality_clustering(g, done, 
+  betweenness_centrality_clustering(g, done,
```
Comment (jzmaddock) [inline include/boost/graph/bc_clustering.hpp]: Another 2 files with just whitespace changes: with over 200 changed files and a mix of substantive and formatting changes this is getting rather hard going.   Please can we split this into smaller orthogonal changes?

Stopping at this point...

**Comment by anadon** [comment]
You can change the review settings to hide files with only whitespace
changes.

**Pull request closed without merging**